libs/src/main/python/dlpx/virtualization/libs/libs.py

```diff
@@ -468,7 +468,10 @@ def retrieve_credentials(credentials_supplier):
     response = internal_libs.retrieve_credentials(credentials_request)
     response_to_str(response)
     credentials_result = _handle_response(response)
-    if credentials_result.password != "":
+    if (
+        credentials_result.key_pair.private_key == "" and
+        credentials_result.key_pair.public_key == ""
```

Comment:
`if not credentials_result.key_pair.private_key and not credentials_result.key_pair.public_key:`?
Don't know if this actually works but ideally we shouldn't be comparing empty strings.

Reply:
I have made the changes as suggested